File: codewars/7kyu/remove-duplicate-words/dduck-remove-duplicate-words.py
# https://www.codewars.com/kata/5b39e3772ae7545f650000fc

from unittest import TestCase

from collections import OrderedDict

remove_duplicate_words = lambda s: " ".join([x for x in OrderedDict.fromkeys(s.split(' '))])

TestCase().assertEqual(
    remove_duplicate_words("alpha beta beta gamma gamma gamma delta alpha beta beta gamma gamma gamma delta"),
    "alpha beta gamma delta")
TestCase().assertEqual(remove_duplicate_words("my cat is my cat fat"), "my cat is fat")

File: codewars/7kyu/remove-duplicate-words/dduck-remove-duplicate-words.js
// https://www.codewars.com/kata/5b39e3772ae7545f650000fc

const dduckRemoveDuplicateWords = (s) => Array.from(new Set(s.split(' '))).join(' ');

console.log(dduckRemoveDuplicateWords('alpha beta beta gamma gamma gamma delta alpha beta beta gamma gamma gamma delta'), 'alpha beta gamma delta');
